### ISSUE TITLE
Indicate what backend is used for config storage in the status page

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#6520]
With #2277 in for the upcoming 1.28.0 release, we should be indicating in the status report page whether files or the db is being used for config. Perhaps links to documentation as well?

**Comment by indigoxela** [comment]
> we should be indicating in the status report page whether files or the db is being used for config

Yes, we definitely should, as this is an important info. @klonos do you have time to provide a PR? It would be cool, if we managed to get this into the 1.28 stable release.

**Comment by klonos** [comment]
> @klonos do you have time to provide a PR? It would be cool, if we managed to get this into the 1.28 stable release.

Yes, of course. Working on it...

**Comment by indigoxela** [comment]
Ah, a change of scope just happened. :wink: 

To be clear: adding this to _telemetry_ would only be useful, it we needed some sort of decision base, which we don't. We just added the ability and won't remove it any time soon.

Adding the info to the **status** page is, what I consider useful here.

**Comment by klonos** [comment]
> Ah, a change of scope just happened. 😉

Fair enough. I'll move telemetry to another issue.

**Comment by klonos** [comment]
Here's a PR: https://github.com/backdrop/backdrop/pull/4739

<img width="801" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/de77db10-d3a4-41a1-9e19-292e9f4138d0">

Given concerns raised in #5791 with regards to not exposing the config directories, I'm only indicating `files` or `database` at the moment. Also, should I be accounting for contrib providing other methods? (in which case, what should we be indicating?)

**Comment by indigoxela** [comment]
Looks good! :+1:

One thing came to my mind... wording:

"Configuration storage locations" - technically it's a "location", but actually, for the admin looking at that page, the "location" would be the path to the directory, if it were files based. What they see is actually the ... hm ... "type"?
Personally, I find the word "location" ambiguous here. :wink: But I don't have strong feelings re that phrasing.

**Comment by klonos** [comment]
How about "Configuration storage backend" or simply "Configuration storage" then? ...other thoughts?

**Comment by indigoxela** [comment]
> How about "Configuration storage backend"

Perfect! :+1:

**Comment by klonos** [comment]
OK, I'll give it some time to allow others to chime in, and I'll then update the PR. In the meantime, what about the strings for the values? ...should it be as it currently is:
- Active configuration: files
- Staging configuration: files
...or simply:
- Active: files
- Staging: files

In the dev meeting this past week, it was also mentioned that we should add some help text as well, with a link to documentation about this. We currently have no documentation for this though. I was thinking something along these lines:
> By default, Backdrop stores its configuration in .json files (in the `/files/config` directory). In certain hosting providers, This can be adjusted in the `settings.php` file.

Thoughts?

**Comment by stpaultim** [comment]
> Given concerns raised in https://github.com/backdrop/backdrop-issues/issues/5791 with regards to not exposing the config directories, I'm only indicating files or database at the moment. Also, should I be accounting for contrib providing other methods? (in which case, what should we be indicating?)

I'm not advocating one way or another, but the concern about sharing complete path to files directory was that the debug info was specifically intended to be shared, potentially in a public forum. I think providing a path the admin on the status page would be different.

**Comment by indigoxela** [comment]
Friendly reminder, that the release providing that new feature will come out in a few days.
If we get into bike shedding here, the accompanying info on status page may not make it in in time. :wink:

**Comment by indigoxela** [comment]
If displaying the config file path (as set in settings.php) is an option, then the text can just stay as-is. Because then it's the "path" that admins would expect, anyway.

@klonos if your time permits, mind to update the PR to actually show the path? I haven't been involved in any discussion re path display, but I guess, for the admins (only privileged accounts may see this page) this info is just fine to show.

I'd really like to get this into 1.28.0, as this is the only "user facing" thing for the config backend. It's info and feature presentation in one. :wink:

**Comment by klonos** [comment]
> only privileged accounts may see this page

Yes, that was the counter-argument I presented in #5791, but people still had concerns. What I think we can do is the same thing that we are doing with the cron job entry, where we are hiding any sensitive information inside a "more" link. I'll update the PR to do that.

Can I please get some suggestions for the help text to add, so that I update everything in one go?

**Comment by indigoxela** [comment]
@klonos I do get the concerns for #5791 and I think, they're valid _there_. But the use-case is different here, as this info is for admins, _not_ to paste to some potentially public space (the purpose of the debug info).

But if there are concerns, your first suggestion "Configuration storage backend" works just great as label.

And even if everything stays as-is in the current PR, I'm fine with it. It feels a bit ambiguous, but is still valuable info.

**Comment by quicksketch** [comment]
I think this PR is good for starters. I also like that we can swap out the words "files" or "database" for directories or database tables if we so desire without changing the translation strings.

**Comment by klonos** [comment]
We still need feedback on the following strings for this:
- `Configuration storage locations` or `Configuration storage backends` or just `Configuration storage`?
- `Active configuration` and `Staging configuration` or just `Active` and `Staging`?
- `files` or `directories` ...or `.json files`? (specifically mention the type of files)
- `database` or `database tables`?

It would be nice to have some documentation available that we can link to (things like "this can only be changed during installation currently" etc.)

**Comment by olafgrabienski** [comment]
>> only privileged accounts may see this page

> Yes, that was the counter-argument I presented in https://github.com/backdrop/backdrop-issues/issues/5791, but people still had concerns.

I raised the concerns about the Debug report, but that is another case because it's optimized for sharing information via copy & paste. In the status report, it's fine to show the information in my opinion.

**Comment by klonos** [comment]
Thanks @olafgrabienski 🙏🏼 ...it seems that we have a few people here to state that this is not a concern in the status report page. I'll work on the PR to add the config paths and db string then.

**Comment by indigoxela** [comment]
>  but what is it important for?

@findlabnet  Core as of 1.28 provides two different config storage options - files or database. The latter is new. Which one the site uses, is important info for admins and should get displayed. At least, as an admin I'd like to know "at a glance", without accessing the settings.php file.

**Comment by findlabnet** [comment]
@indigoxela thanks for the clarification, sorry for the wrong question, I was thinking of something else.

**Comment by jenlampton** [comment]
Here are my votes for things :)

> We still need feedback on the following strings for this:

> Configuration storage locations or Configuration storage backends or just Configuration storage?

I vote for "Configuration storage" (will look better on smaller screens)

> Active configuration and Staging configuration or just Active and Staging?

I vote for "Active" and "Staging"

> files or directories ...or .json files? (specifically mention the type of files)

I like "json files"

> database or database tables?

I like "database"

**Comment by klonos** [comment]
Thanks @jenlampton 🙏🏼 ...since nobody else provided any feedback with regards to these for some time, I'll go ahead with your suggestions. I'll update the PR, but happy to adjust if others come and ask for tweaks.

**Comment by klonos** [comment]
> ...and think the bulleted list is not needed.

@izmeez bulleted lists are added for a11y/readability when presenting lists of items. They also provide semantic meaning to the HTML added to the page for the respective elements/items.

> The code also tweaks comments including in the settings.php file that are not specific to this but incidental fixes.

Sure 👍🏼 ...I've opened #6612 for that.

**Comment by klonos** [comment]
OK, PR updated.

@izmeez I've reverted the changes in settings.php, but I've also refactored things a bit, so this will need a second round of testing please.

@jenlampton the `'active'` and `'staging'` array keys need to be translated individually, and since they are dynamically changed within the `foreach`, I had to use the odd `t($variable)` thing. Later, when calling the actual `t()` for the entire string, I'm using `!` instead of `@` for this specific replacement. I hope that this is done properly now.

**Comment by kiamlaluno** [comment]
There is no need to pass a variable to `t()`: It is sufficient to use code similar to the following one.

```php
$config_storage_types = array('active' => t('Active'), 'staging' => t('Staging'));
foreach ($config_storage_types as $type => $type_name) {
  // …
}
```

**Comment by klonos** [comment]
Thanks @kiamlaluno 🙏🏼 ...it does make sense 👍🏼 ...PR updated.

@izmeez that simpler format is what we should add to the debug info at `admin/reports/debug`. Others have requested that in the site status report they also expect to see the path to the config or the db/table info. See comments [here](https://github.com/backdrop/backdrop-issues/issues/6520#issuecomment-2106096933) and [here](https://github.com/backdrop/backdrop-issues/issues/6520#issuecomment-2109730407), and after that.

**Comment by stpaultim** [comment]
I like what I see here:
![image](https://github.com/backdrop/backdrop-issues/assets/3144571/ef51d68f-e3cb-4ddc-adea-815084214b67)

**Comment by kiamlaluno** [comment]
For a moment, I thought that 40.2.0 was the Backdrop core version. 😳😅

**Comment by quicksketch** [comment]
I retested this locally and in the PR sandbox. The code seems a little complicated for getting such simple output but it does the job. I'm also not 100% about using a `<ul>`, since we're not doing that for any other status report items when viewing a fresh install. But in all, I think this provides value as-is and we don't need to go around in circles trying to address my (and others) every nitpick. I merged https://github.com/backdrop/backdrop/pull/4739 into 1.x and 1.28.x. Thanks everyone!

[ed13104](https://github.com/backdrop/backdrop/commit/ed13104335828e3c1c10f04f3c1adf92b7569285) by @klonos, @indigoxela, @izmeez, @stpaultim, @jenlampton, @kiamlaluno, @olafgrabienski, and @findlabnet.
